lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -95,7 +95,8 @@ class MongoClientInterface {
             isLocationTransient } = params;
         const cred = MongoUtils.credPrefix(authCredentials);
         this.mongoUrl = `mongodb://${cred}${replicaSetHosts}/` +
-            `?w=${writeConcern}&replicaSet=${replicaSet}` +
+            `?w=${writeConcern}` +
+            `&${replicaSet !== undefined ? 'replicaSet=' : ''}${replicaSet}` +
```

Comment:
this creates a malformed mongourl if `replicaSet` is undefined/null/empty.

```suggestion
            `${!!replicaSet ? '&replicaSet='+replicaSet : ''}` +
```

Reply:
yes right! thanks